### ISSUE TITLE
ALS bomb upon launch from #849 has reappeared in version 23.0.2

### ISSUE THREAD

**Issue by oscarrutt** [AdaCore/ada_language_server#875]
Sorry to report that the problem of ALS bombing upon launching Visual Studio Code, as in #849, has reappeared in ALS version 23.0.2. I have reverted to 23.0.1 by swapping in the special build linked from #849.

VSC "About:"

Version: 1.63.0
Commit: 7db1a2b88f7557e0a43fec75b6ba7e50b3e9f77e
Date: 2021-12-07T05:15:48.091Z
Electron: 13.5.2
Chromium: 91.0.4472.164
Node.js: 14.16.0
V8: 9.1.269.39-electron.0
OS: Darwin x64 15.6.0

**Comment by reznikmm** [comment]
I found `___darwin_check_fd_set_overflow` symbol in `process-wrappers.o` which is part of GNATCOLL. It looks like ALS doesn't use this part of GNATCALL. It's unclear to my why linker includes unneeded objects into a statically linked executable. I'll try to drop this file before ALS is compiled to see if this fixes the issue.

I've tried to set `MACOSX_DEPLOYMENT_TARGET=10.11` environment and `-Wl,-no_weak_imports` linker option, but it doesn't help much, as far as I can see.

I don't have an access to Mac OS X machine, so the investigation is hard.
